<strong>Доказать двумя способами, что оператор $Ф: C[a; b] \to C[a; b]$ является сжимающим:
    
A. по определению сжимающего оператора;

B. по достаточному признаку сжимающего оператора.</strong>

$Ф[x] = \frac{1}{2 + x^2(t)} - \sqrt{t}, [a; b] = [0; 1]$

<i>A. Обратимся непосредственно к определению сжимающего оператора.</i>

$\rho_{C[0; 1]}(Ф[x], Ф[y]) = \max_{t \in [0; 1]}{|Ф[x] - Ф[y]|} = \max_{t \in [0; 1]}|\frac{1}{2 + x^2(t)} - \frac{1}{2 + y^2(t)}| = \max_{t \in [0; 1]}|\frac{y^2(t) - x^2(t)}{(2 + x^2(t))(2 + y^2(t))}| = \max_{t \in [0; 1]}|y(t) - x(t)| \max_{t \in [0; 1]}|\frac{y(t) + x(t)}{(2 + x^2(t))(2 + y^2(t))}| =  \max_{t \in [0; 1]}|\frac{y(t) + x(t)}{(2 + x^2(t))(2 + y^2(t))}|
\rho_{C[0; 1]}(x, y)$

Найдем $\max_{t \in [0; 1]}|\frac{y(t) + x(t)}{(2 + x^2(t))(2 + y^2(t))}|$.
Для этого рассмотрим функцию двух переменных $\frac{y(t) + x(t)}{(2 + x^2(t))(2 + y^2(t))}$:

In [25]:
import sympy as sp
from IPython.display import display

sp.init_printing(use_latex='mathjax')

a = sp.Symbol('a')
b = sp.Symbol('b')
expr = (a + b)/((2 + a**2)*(2 + b**2))
display(expr)

da = expr.diff(a)
db = expr.diff(b)
solve_da = sp.solve(da)
b1 = sp.solve(db.subs(a, solve_da[0][a]))[0]
b2 = sp.solve(db.subs(a, solve_da[1][a]))[0]
a1 = solve_da[0][a].subs(b, b1)
a2 = solve_da[1][a].subs(b, b2)

display(a1)
display(a2)
display(b1)
display(b2)

      a + b      
─────────────────
⎛ 2    ⎞ ⎛ 2    ⎞
⎝a  + 2⎠⋅⎝b  + 2⎠

-√6 
────
 3  

√6
──
3 

-√6 
────
 3  

√6
──
3 

Теперь найдем частные производные второго порядка:

In [9]:
da2 = da.diff(a)
db2 = db.diff(b)
dab = da.diff(b)

display(da2)
display(db2)
display(dab)

      2                                                     
   8⋅a ⋅(a + b)             4⋅a               2⋅(a + b)     
────────────────── - ────────────────── - ──────────────────
        3                    2                    2         
⎛ 2    ⎞  ⎛ 2    ⎞   ⎛ 2    ⎞  ⎛ 2    ⎞   ⎛ 2    ⎞  ⎛ 2    ⎞
⎝a  + 2⎠ ⋅⎝b  + 2⎠   ⎝a  + 2⎠ ⋅⎝b  + 2⎠   ⎝a  + 2⎠ ⋅⎝b  + 2⎠

      2                                                     
   8⋅b ⋅(a + b)             4⋅b               2⋅(a + b)     
────────────────── - ────────────────── - ──────────────────
                 3                    2                    2
⎛ 2    ⎞ ⎛ 2    ⎞    ⎛ 2    ⎞ ⎛ 2    ⎞    ⎛ 2    ⎞ ⎛ 2    ⎞ 
⎝a  + 2⎠⋅⎝b  + 2⎠    ⎝a  + 2⎠⋅⎝b  + 2⎠    ⎝a  + 2⎠⋅⎝b  + 2⎠ 

   4⋅a⋅b⋅(a + b)             2⋅a                  2⋅b        
─────────────────── - ────────────────── - ──────────────────
        2         2           2                             2
⎛ 2    ⎞  ⎛ 2    ⎞    ⎛ 2    ⎞  ⎛ 2    ⎞   ⎛ 2    ⎞ ⎛ 2    ⎞ 
⎝a  + 2⎠ ⋅⎝b  + 2⎠    ⎝a  + 2⎠ ⋅⎝b  + 2⎠   ⎝a  + 2⎠⋅⎝b  + 2⎠ 

Вычислим значение этих частных производных второго порядка в критических точках $M(x_0;y_0)$:

In [29]:
# Первая точка
r1 = da2.subs(a, a1).subs(b, b1)
r2 = db2.subs(a, a1).subs(b, b1)
r3 = dab.subs(a, a1).subs(b, b1)

display(r1*r2 - r3**2)
display(r1)
# Первая точка
r1 = da2.subs(a, a2).subs(b, b2)
r2 = db2.subs(a, a2).subs(b, b2)
r3 = dab.subs(a, a2).subs(b, b2)

display(r1*r2 - r3**2)
display(r1)

 729 
─────
32768

9⋅√6
────
128 

 729 
─────
32768

-9⋅√6 
──────
 128  

Следовательно, $(\frac{\sqrt{6}}{3}; \frac{\sqrt{6}}{3})$ - точка максимума, а $(- \frac{\sqrt{6}}{3}; - \frac{\sqrt{6}}{3})$ - точка минимуму функции $\frac{y(t) + x(t)}{(2 + x^2(t))(2 + y^2(t))}$.

Теперь уже рассмотрим максимум функции $|\frac{y(t) + x(t)}{(2 + x^2(t))(2 + y^2(t))}|$: в точке $(\frac{\sqrt{6}}{3}; \frac{\sqrt{6}}{3})$

In [26]:
a_max = 6**(0.5)/3
b_max = 6**(0.5)/3
display(expr.subs(a, a_max).subs(b, b_max))

0.229639663385923

$\rho_{C[0; 1]}(Ф[x], Ф[y]) \leq 0.229639663385923 \rho_{C[0; 1]}(x, y)$

Приходим к выводу, что оператор $Ф$ сжимающий с коэффициентом сжатия $\alpha = 0.229639663385923$

<i>B. Обратимся к достаточному признаку сжимающего оператора в пространстве непрерывных функций.</i>

Обозначим $Ф[x] = \varphi(t, x(t))$, где $\varphi(t, u) = \frac{1}{2 + u} - \sqrt{t}$.
Очевидно, что функция двух переменных $\varphi(t, u)$ непрерывна на множестве $[0; 1] \times [0; \infty)$, непрерывно дифференцируема по переменной $u$ причем при всех $(t, u) \in [0; 1] \times [0; \infty)$ справедлива оценка:

In [35]:
t = sp.Symbol('t')
u = sp.Symbol('u')

expr = 1/(2 + u**2) - sp.sqrt(t) 

diffed_expr = sp.Abs(expr.diff(u))
display(diffed_expr)
display(sp.solve(expr.diff(u).diff(u)))

r1 = diffed_expr.subs(u, 6**(0.5)/3)
display(r1)

  │    u    │
2⋅│─────────│
  │        2│
  │⎛ 2    ⎞ │
  │⎝u  + 2⎠ │

⎡-√6   √6⎤
⎢────, ──⎥
⎣ 3    3 ⎦

0.229639663385923

$|\varphi^{'}_u(t, u)| = 2 \left|{\frac{u}{\left(u^{2} + 2\right)^{2}}}\right| \leq 0.229639663385923 < 1$.

Значит, оператор $Ф$ сжимающий с коэффициентом сжатия $\alpha = 0.229639663385923$.